tests/baselines/reference/typeGuardTypeOfUndefined.types

```diff
@@ -367,15 +367,15 @@ function test14(a: boolean | number | void) {
         if (typeof a === "boolean") {
 >typeof a === "boolean" : boolean
 >typeof a : "string" | "number" | "boolean" | "symbol" | "undefined" | "object" | "function"
->a : number | boolean | void
+>a : undefined

```

Comment:
This is kind of a scary diff - is there an observable breaking change we need to document?

Reply:
`a` started as `boolean | number | void` - it then for narrowed against `undefined` (which actually matches the `void` case - and `void & undefined` is apparently just `undefined`). Previously we just _missed_ that narrowing opportunity. And naturally, `undefined` narrowed by `boolean` simply becomes `never` below, like one would hope!